### PR TITLE
add Dockerfile and buildspec for nodejs4.3

### LINKED ISSUE #11: Support nodejs4.3 runtime
Lambda supports nodejs4.3, we should support it too.

### PR DESCRIPTION
**Pull request by emilymcafee**
Closes #11 

- Got node + npm versions from what is running on a nodejs4.3 function
- Copied everything else from the node6.x installation
- Will clean up README in another pass

I believe 6.x will still be the default, and 4.3 functions will have to ask for that explicitly.

@rclark is there anything I'm missing here? Tests / anything else?

### CODE REVIEW

File: Dockerfiles/nodejs4.3 [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM amazonlinux
+
+RUN yum install -y wget zip
+RUN curl -sL https://nodejs.org/dist/v4.3.2/node-v4.3.2-linux-x64.tar.gz | tar zxC /usr/local --strip-components=1
+RUN rm -rf /usr/local/lib/node_modules/npm && \
+  mkdir /usr/local/lib/node_modules/npm && \
+  curl -sL https://github.com/npm/npm/archive/v2.14.12.tar.gz | tar xz -C /usr/local/lib/node_modules/npm --strip-components=1
```
Comment (rclark) [inline Dockerfiles/nodejs4.3]: In the `6.x` image, all this npm junk is just about getting npm 5.3.0 onto the image. This means that stork will build the bundle respecting a `package-lock.json` file if the repository being bundled has one. 

I think we should also install v5.3.0 here.

**Review by rclark — COMMENTED** [review]
The URI for the image will need to be [added to this array](https://github.com/mapbox/stork/blob/8bae707a3f0ce3d3d14e49cebd2d9c0dd7c99c2e/lambda.js#L265) in the lambda function code.

**Comment by rclark** [comment]
If you want to write a test, [you can more or less duplicate this one](https://github.com/mapbox/stork/blob/8bae707a3f0ce3d3d14e49cebd2d9c0dd7c99c2e/test/lambda.test.js#L335), but [change the mocked representation of the `.stork.json` file](https://github.com/mapbox/stork/blob/8bae707a3f0ce3d3d14e49cebd2d9c0dd7c99c2e/test/lambda.test.js#L342-L347) so that it asks for node 4.3, and change the assertions about the project that was created.

File: Dockerfiles/nodejs4.3 (hunk 1 above)
Comment (emilymcafee) [reply Dockerfiles/nodejs4.3]: 👍

**Pull request merged**